packages/synapse-react-client/src/lib/containers/CardContainerLogic.tsx

```diff
@@ -159,7 +160,7 @@ export const CardContainerLogic = (props: CardContainerLogicProps) => {
     entityId: entityId,
     query: {
       sql: props.sql,
-      limit: props.limit,
+      limit: props.limit ?? DEFAULT_PAGE_SIZE,
```

Comment:
This is a good place for the fix. 

I think we should also fix the hook, since this would happen again if a new component used it and didn't fall back to the default limit. I wonder if it's as simple as changing `getNextPageParam` to use the number of rows for page size.

```tsx
// currently:
const pageSize = request.query.limit ?? DEFAULT_PAGE_SIZE
// suggested change
const pageSize =
  request.query.limit ??
  page.responseBody!.queryResult?.queryResults.rows.length ??
  DEFAULT_PAGE_SIZE
// or maybe just this
const pageSize = page.responseBody!.queryResult?.queryResults.rows.length ?? DEFAULT_PAGE_SIZE
```

I would be fine with having this suggested change tracked in a new ticket so we could dedicate a bit of time to test it properly, since there might be a few edge cases to think about.

Thoughts?

Reply:
I think this could result in an infinite loop.  If the row length is 0 (and the request query limit is undefined), then `getNextPageParam` will still return true (pageSize === row count)